arch/BPF/BPFInstPrinter.c

```diff
@@ -217,22 +239,35 @@ static void print_operand(MCInst *MI, struct SStream *O, const cs_bpf_op *op)
 		SStream_concat(O, BPF_reg_name((csh)MI->csh, op->reg));
 		break;
 	case BPF_OP_IMM:
-		SStream_concat(O, "0x%" PRIx64, op->imm);
+		if (op->is_signed && ((int32_t)op->imm) < 0)
+			SStream_concat(O, "-0x%" PRIx64, -(int32_t)op->imm);
+		else
+			SStream_concat(O, "0x%" PRIx64, op->imm);
```

Comment:
> If you think the numbers should always be in hex format (no matter what), please add `printInt64Hex` etc. functions in `SStream.h`.

Yep I'll add, but anyway I think we should rewrite some of them, because there is a lot of code duplication there and the functions themselves are very niche

Reply:
Maybe in the future. But it has low priority for now. Too many other construction sides.